File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int l = 0, j = 1;
        while(j<prices.length) {
            if(prices[l]<prices[j])
                maxProfit = Math.max(maxProfit, prices[j] - prices[l]);
            else
                l = j;
            j++;
        }
        return maxProfit;
    }

    
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices: return 0

        minPrice, maxProfit = prices[0], 0
        for price in prices:
            minPrice = min(minPrice, price)
            maxProfit = max(maxProfit, price - minPrice)
        return maxProfit